tools/proofers/preview.inc

```diff
@@ -71,25 +71,20 @@ function output_preview_strings()
     $demo_string = "$plain _{{$subscript}} <i>$ss[italic]</i> ^{{$superscript}} <b>$ss[bold]</b> <g>$ss[gesperrt]</g> <sc>$ss[small_caps]</sc> <f>$ss[font_change]</f>";
     $ie_warn = javascript_safe(_("This function is not available for Internet Explorer versions less than 9"), $charset);
 
-
-    // json_encode() assumes everything is in UTF-8, but we may not be so
-    // just build the JSON object ourselves
-    $json_entries = [];
-    foreach($messages as $key => $value)
+    if(!$utf8_site)
     {
-        $value = javascript_safe($value);
-        $json_entries[] = "\"$key\": \"$value\"";
+        array_walk($messages, function(&$item) { $item = utf8_encode($item);} );
     }
-    $preview_msg = "{" . implode(",", $json_entries) . "}";
+    $preview_msg = json_encode($messages);
 
-    echo "var previewMessages = JSON.parse('$preview_msg'); var previewDemo = '$demo_string'; var ieWarn = '$ie_warn'";
+    return "var previewMessages = $preview_msg; var previewDemo = '$demo_string'; var ieWarn = '$ie_warn';";
```

Comment:
We should run `javascript_safe()` on `$demo_string` and `$ie_warn` in case the translation contains a single quote.

Reply:
I think it is, but the code is written in such a strange way it's not obvious. Who wrote this stuff? Oh wait ...
I'll perhaps rewrite it.

Reply:
Ah, yes, this is more straightforward - thanks.